File: examples/java/sdk-reference/observability/logger-interface.java
import org.slf4j.Logger;
import software.amazon.lambda.durable.logging.DurableLogger;

// DurableLogger wraps an SLF4J Logger and adds MDC entries:
//   durableExecutionArn, requestId, operationId, operationName, attempt
//
// Obtain it from any context:
//   DurableLogger logger = context.getLogger();
//   DurableLogger logger = stepContext.getLogger();
//
// Available methods:
//   logger.trace(String format, Object... args)
//   logger.debug(String format, Object... args)
//   logger.info(String format, Object... args)
//   logger.warn(String format, Object... args)
//   logger.error(String format, Object... args)
//   logger.error(String message, Throwable t)

File: examples/typescript/sdk-reference/observability/logger-interface.ts
import { DurableLogger, DurableLoggingContext } from "@aws/durable-execution-sdk-js";

// DurableLogger interface — implement this to use a custom logger.
interface DurableLogger {
  log?(level: "INFO" | "WARN" | "ERROR" | "DEBUG", ...params: unknown[]): void;
  error(...params: unknown[]): void;
  warn(...params: unknown[]): void;
  info(...params: unknown[]): void;
  debug(...params: unknown[]): void;
  configureDurableLoggingContext?(context: DurableLoggingContext): void;
}
